src/components/KCatalog/KCatalog.vue

```diff
@@ -393,11 +423,35 @@ const fetchData = async () => {
     query: searchInput || filterQuery.value,
     pageSize: pageSize.value,
     page: page.value,
+    offset: offset.value,
   })
 
   data.value = res.data as CatalogItem[]
   total.value = props.paginationTotalItems || res.total || res.data?.length
 
+  if (props.paginationOffset) {
+    if (!res.pagination?.offset) {
+      nextOffset.value = null
+    } else {
+      nextOffset.value = res.pagination.offset
+
+      if (!offsets.value[page.value]) {
```

Comment:
same question, need to make sure `offsets.value` is defined before checking for a specific page